syntax = "proto3";

package user;

service User {
   rpc Signup(SignupRequest) returns (SignupResponse) {}
   rpc Signin(SigninRequest) returns (SigninResponse) {}
   rpc ChangePassword(ChangePasswordRequest) returns (ChangePasswordResponse) {}
   rpc UpdateUserProfile(UpdateUserProfileRequest) returns (UpdateUserProfileResponse) {}
   rpc AddRole(AddRoleRequest) returns (AddRoleResponse) {}
   rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse) {}
   rpc UploadFile(UploadFileRequest) returns (UploadFileResponse) {}
   rpc UserBoxerProfile(UserBoxerProfileRequest) returns (UserBoxerProfileResponse) {}
//   rpc Me(MeRequest) returns (MeResponse) {}
//   rpc Boxers() returns () {}
//   rpc Judges() returns () {}
//   rpc Coaches() returns () {}
//   rpc Organizers() returns () {}
}

message UserBoxerProfileRequest {
  string user_id = 1;
}

message UserBoxerProfileResponse {
  string first_name = 1;
  string last_name = 2;
  string sex = 3;
  string birthday = 4;
  string country = 5;
  string region = 6;
  double weight = 7;
  double height = 8;
  string athletic_distinction = 9;
  string avatar = 10;
}

//message MeRequest {
//  string user_id = 1;
//}
//
//message MeResponse {
//  string first_name = 1;
//  string last_name = 2;
//  string middle_name = 3;
//  string email = 4;
//  string phone_number = 5;
//  string sex = 6;
//  string birthday = 7;
//  string country = 8;
//  string region = 9;
//  string city = 10;
//  string weight = 11;
//  string height = 12;
//  string athletic_distinction = 13;
//  bytes avatar = 14;
//}

message UploadFileRequest {
  bytes file_content = 1;
  bool is_avatar = 2;
  string content_type = 3;
  string user_id = 4;
}

message UploadFileResponse {
  string message = 1;
}

message SignupRequest {
   string account_type = 1;
   optional string phone_number = 2;
   optional string email = 3;
   string password = 4;
   string password_confirm = 5;
}

message SignupResponse {
   string message = 1;
}

message SigninRequest {
   optional string email = 1;
   optional string phone_number = 2;
   string password = 3;
}

message SigninResponse {
   string token_type = 1;
   string access_token = 2;
   string refresh_token = 3;
}

message ChangePasswordRequest {
   string current_password = 1;
   string new_password = 2;
   string new_password_confirm = 3;
   string user_id = 4;
}

message ChangePasswordResponse {
   string token_type = 1;
   string access_token = 2;
   string refresh_token = 3;
}

message UpdateUserProfileRequest {
   optional string first_name = 1;
   optional string last_name = 2;
   optional string sex = 3;
   optional string middle_name = 4;
   optional string birthday = 5;
   optional string country = 6;
   optional string region = 7;
   optional string city = 8;
   string user_id = 9;
}

message UpdateUserProfileResponse {
   string message = 1;
}

message AddRoleRequest {
   string account_type = 1;
   string user_id = 2;
}

message AddRoleResponse {
   string message = 1;
}

message DeleteUserRequest {
   string user_id = 1;
}

message DeleteUserResponse {
   string message = 1;
}

